src/widgets/barchart/mod.rs

```diff
@@ -756,4 +835,44 @@ mod tests {
         let expected = Buffer::with_lines(vec!["  █", "█ █", " G "]);
         assert_buffer_eq!(buffer, expected);
     }
+
+    #[test]
+    fn test_horizontal_bars() {
```

Comment:
Imagine the next person to implements a new feature in barchart makes this test fail. One of the problems with writing tests that test all the features at once is that it requires any failure to now fully understand the entire code being tested rather than just the part that failed.

One way to help that person (which could even be your future self) would be to forget about the implementation for a minute and write a list of the things that this test tells you about how the implementation works (and why).

For instance what is the significance to the number 5 in the first bar?

Is there a reason that the bars are rendered in reverse order from their definition?

Is there a reason why the group name is the part that is not rendered rather than one of the bars?

How is each edge case handled? What are they?

Reply:
true. I was lazy :smiley_cat: . I will rework this

Reply:
> For instance what is the significance to the number 5 in the first bar?

has no significance. all I need a bi bar and a small one, which doesn't fit the text

> Is there a reason that the bars are rendered in reverse order from their definition?

I guess, you mean that the first bar is rendered at the bottom. I think looks nicer. usually graphs, bars, charts start always from the bottom

> Is there a reason why the group name is the part that is not rendered rather than one of the bars?

because there is no test case for group name outside the scope.  On the other hand bars outside the scope is already tested in the old code and still valid  for this MR, because the function responsible for removing bars outside the scope is `remove_invisible_groups_and_bars`. It is a common function for the horizontal and vertical rendering.

> How is each edge case handled? What are they?

I added tests for the styles. I hope it enough :smile_cat: